File: basic/string/longestCommonPrefix/longestCommonPrefix.cpp
	string longestCommonPrefix(vector<string>& str){
			//your code goes here
            if(str.empty()) return "";
            int n=str.size();
            int m=str[0].size();
            sort(str.begin(),str.end());
            string first=str[0];
            string last=str[n-1];
            string ans="";
            for(int i=0;i<min(first.size(),last.size());i++){
                if(first[i]!=last[i]) break;
                ans+=first[i];
            }
            return ans;

		}

File: basic/string/longestCommonPrefix/longestCommonPrefix.py
    def longestCommonPrefix(self, st):
        #your code goes here
        if not st:
            return ""
       
        for i in range(len(st[0])):
            commonstr=st[0][i]
            for j in range(1,len(st)):
                if i>=len(st[j]) or st[j][i]!=commonstr:
                    return st[0][:i];      
        return st[0];  